### ISSUE TITLE
Implement Norm-Adaptive Margins for Hyperbolic Loss

### ISSUE THREAD

**Issue by lowmason** [lowmason/naics-embedder#24]
### Description

The Sampling Strategy PDF recommends norm-adaptive margins to account for hyperbolic volume expansion. While logging exists for adaptive margins, the actual margin computation doesn't appear to use the recommended formula.

### Current Behavior

Metrics are logged (`train/curriculum/adaptive_margin_mean`) but the loss function uses a fixed temperature/margin approach.

### Expected Behavior

Implement norm-adaptive margin from Section 6.3:

```python
def compute_adaptive_margin(self, anchor_emb, base_margin=0.5):
    """
    Compute norm-adaptive margin based on anchor's hyperbolic radius.
    
    In the Lorentz model, density increases exponentially with distance from origin.
    A fixed margin at the root (sparse) is too large at the leaves (dense).
    
    Formula: m(a) = m_0 · sech(||a||_L)
    """
    # Compute Lorentz norm (hyperbolic radius) - this is x_0 for Lorentz model
    lorentz_norm = anchor_emb[:, 0]  # Time coordinate
    
    # sech(x) = 1/cosh(x) = 2/(e^x + e^-x)
    adaptive_margin = base_margin * (1.0 / torch.cosh(lorentz_norm))
    
    return adaptive_margin
```

Then integrate into loss:
```python
# In triplet-style formulation
margin = self.compute_adaptive_margin(anchor_emb)
loss = torch.max(torch.zeros_like(pos_dist), pos_dist - neg_dist + margin)
```

### Rationale

From the Sampling Strategy PDF:
> "In the Lorentz model, the 'density' of space increases exponentially with distance from the origin. A fixed margin m that works at the root (sparse) will be too large at the leaves (dense)."

This ensures tight clusters at the leaves (fine-grained industries) without enforcing impossible constraints at the root.

### Acceptance Criteria

- [x] Implement `compute_adaptive_margin()` method using sech formula
- [x] Add `base_margin` hyperparameter to config
- [x] Integrate with DCL loss or add separate margin-based loss term
- [x] Log margin statistics per batch
- [x] Add tests comparing fixed vs adaptive margin behavior

### Related Files

- `src/naics_embedder/text_model/loss.py`
- `src/naics_embedder/utils/config.py`

---
